### ISSUE TITLE
is there any PyTorch implementation?

### ISSUE THREAD

**Issue by alibool** [andreped/livermask#14]

**Comment by andreped** [comment]
No, but it should not be that relevant as this is just an inference tool. However, it is simple to convert the model to ONNX, which can be deployed from PyTorch. The actual inference part is only a small part of the whole pipeline.

But I would not spent time on that, unless you are interested in integrating the liver segmentation part directly into some other framework based on pytorch. Is That what you are trying to do?

**Comment by alibool** [comment]
thanks for your reply, and yes. If you open source the training code, I am willing to implement the pytorch counterpart.

**Comment by andreped** [comment]
The training of the model used in this project was extremely simple (basic 2D U-Net), and I did not think about making it sharable with others. I therefore no longer have access to the code. However, if you wish to train your own models, and you want to use pytorch, I would strongly recommend using [MONAI](https://github.com/Project-MONAI/MONAI). 

MONAI has all the components you need for training models and deploying them, specifically for medical data. The framework also contains a lot of different methods relevant for 3D medical data (preprocessing of CT/MRI, patching of data, one-hot encoding, loss functions, data augmentations... You name it!). If you wish to train your own model using MONAI, I would also suggest using a 3D architecture, such as the SWIN UNETR, which is easily accessible in MONAI (information regarding the architecture can be found [here](https://developer.nvidia.com/blog/novel-transformer-model-achieves-state-of-the-art-benchmarks-in-3d-medical-image-analysis/)).

Lastly, if you just want to train a model, without going through all the hassle of implementing all the training code yourself and all that. I would recommend [MONAI-Label](https://github.com/Project-MONAI/MONAILabel). The software enables you to train models (and annotate, but which is not relevant in your use case) fully through a GUI, without the need to code. It also has support for the UNETR architecture. It is also extremely easy to setup. Handles everything for you from NIFTI to preprocessing to training!

A demo video of MONAI-Label can be found on [YouTube](https://www.youtube.com/watch?v=o8HipCgSZIw&t=1319s&ab_channel=ProjectMONAI). If you have any questions regarding the software, feel free to ask me. Otherwise, just make an issue in the MONAI-Label repository. They are really fast to reply and make a great effort to assist you in your endevours.

If you have any other questions regarding architecture design, which training setup to use, and whatnot. Feel free to ask me. If you are considering doing a publication, I am also open to collaborate.

Lastly, if you end up with a model that greatly outperforms my very simple livermask tool (should not be that hard tbh). It would be great if you would keep in touch, as I would be very interested in improving the parenchyma segmentation (and vessels) in my tool.

I hope you found this helpful. Best of luck! :]

**Comment by alibool** [comment]
Thanks!
